components/service/firefox-accounts/src/test/java/mozilla/components/service/fxa/FxaAccountManagerTest.kt

```diff
@@ -1886,6 +1887,180 @@ class FxaAccountManagerTest {
         verify(syncManager, times(2)).stop()
         verifyNoMoreInteractions(syncManager)
     }
+    @Test
+    fun `beginAuthenticationAuthSync handles any pairingUrl correctly`() {
+        // Initialize things so the test can run
+        WorkManagerTestInitHelper.initializeTestWorkManager(testContext)
+
+        // Build our dependencies for creating an account manager
+        val tServerConfig = ServerConfig(
+                "https://localhost/contentUrl",
+                "testClientId",
+                "https://localhost/redirectUrl"
+        )
+        val tDeviceConfig = DeviceConfig(
+                "testDevice",
+                DeviceType.MOBILE,
+                setOf()
+        )
+        val tSyncConfig = SyncConfig(setOf(SyncEngine.Bookmarks))
+        val manager = FxaAccountManager(
+                testContext,
+                serverConfig = tServerConfig,
+                deviceConfig = tDeviceConfig,
+                syncConfig = tSyncConfig
+        )
+        val pairingUrl = "https://localhost/pairingUrl".toString()
+        val beginResponse = manager.beginAuthenticationAsync(pairingUrl)
+        assertTrue(beginResponse is CompletableDeferred<String?>)
+    }
+
+    @Test
+    fun `beginAuthenticationAsync with null URL handled correctly`() {
+        // Initialize things so the test can run
+        WorkManagerTestInitHelper.initializeTestWorkManager(testContext)
+
+        // Build our dependencies for creating an account manager
+        val tServerConfig = ServerConfig(
+                "https://localhost/contentUrl",
+                "testClientId",
+                "https://localhost/redirectUrl"
+        )
+        val tDeviceConfig = DeviceConfig(
+                "testDevice",
+                DeviceType.MOBILE,
+                setOf()
+        )
+        val tSyncConfig = SyncConfig(setOf(SyncEngine.Bookmarks))
+        val manager = FxaAccountManager(
+                testContext,
+                serverConfig = tServerConfig,
+                deviceConfig = tDeviceConfig,
+                syncConfig = tSyncConfig
+        )
+        val beginResponse = manager.beginAuthenticationAsync(null)
+        assertTrue(beginResponse is CompletableDeferred<String?>)
+    }
+
+    @Test
+    fun `integration test with latest auth state matching current auth state`() {
```

Comment:
We already have tests which assert that state validation is performed correctly. Please see `auth state verification while finishing authentication` and neighbouring tests.

Tests below are incomplete (same comments about `await`, etc) and appear to be mostly redundant to me?

If existing tests aren't covering cases you're thinking of, please modify them instead of adding entirely new tests.